src/main/java/core/components/GridBoard.java

```diff
@@ -178,7 +171,7 @@ public T getElement(int x, int y) {
      * @param orientation - int orientation, how many times it should be rotated clockwise
      * @return - new grid with the same elements and correct orientation.
      */
-    public T[][] rotate(int orientation) {
+    public Component[][] rotate(int orientation) {
```

Comment:
And see previous point. As a user of GridBoard, I woul dfind it much more natural to use if it returns T[][] from all these other methods. All I'm going to do as a user is cast them to T in the client code anyway!